File: Python/0392_is_subsequence/solution.py
def is_subsequence(s,t):
    result = str()
    current = 0
    for letter in t:
        if current < len(s) and letter == s[current]:
            result = result + letter
            current = current + 1
    if result == s:
        return True
    return False

s = "abc"
t = "ahbgdc"
print(is_subsequence(s,t))

File: JavaScript/0392_is_subsequence/solution.js
function is_subsequence(s,t) {
    let result = new String();
    let current = 0;
    for (let letter of t) {
        if (current < s.length && letter == s[current]) {
            result = result + letter;
            current ++;
        }
    }
    if (result == s) {
        return true;
    }
    return false;
}

const s = "abc";
const t = "ahbgdc";
console.log(is_subsequence(s,t));